### DISCUSSION TITLE
How to reuse mappings like Automapper `IncludeBase`?

### DISCUSSION THREAD

**Discussion by mcliment** [riok/mapperly#1126] [Q&A]
We have quite an amount of AM mappings and we want to migrate to Mapperly, moslty because the code generation features that make mappings much faster and debuggable.

We make heavy use of AM https://docs.automapper.org/en/stable/Mapping-inheritance.html which is quite convenient in our case because: 1) the db model has weird names that we can't change and we have to make most mappings explicit and 2) there are many entities that share the same interface but the different implementations have additional information to map

I've been reading the docs but I don't see how to have something similar to avoid duplicating too many `[MapProperty]`.

Any suggestions?

**Comment by latonz** [comment]
This is not possible yet. I think https://github.com/riok/mapperly/issues/513 could help here. Would that API address your use-case?

**Reply by mcliment** [reply]
Actually #513 looks promising. In the end it's just reusing some code to avoid having too many duplicates, that will help for sure!
